supplementary/alica_ros1/alica_tests/src/test/test_success.cpp

```diff
@@ -0,0 +1,110 @@
+#include <alica_tests/ConditionCreator.h>
+#include <alica_tests/ConstraintCreator.h>
+#include <alica_tests/PlanCreator.h>
+#include <alica_tests/TestWorldModel.h>
+#include <alica_tests/TransitionConditionCreator.h>
+#include <alica_tests/UtilityFunctionCreator.h>
+#include <clock/AlicaRosTimer.h>
+#include <communication/AlicaDummyCommunication.h>
+#include <logger/AlicaRosLogger.h>
+
+#include <gtest/gtest.h>
+#include <ros/ros.h>
+
+namespace alica::test
+{
+
+class TestSuccessFixture : public ::testing::Test
+{
+public:
+    virtual void SetUp() override
+    {
+        ros::NodeHandle nh;
+        std::string path;
+        nh.param<std::string>("/rootPath", path, ".");
+        _tc = std::make_unique<TestContext>("hairy", path + "/etc/", "Roleset", "TestMasterPlan", true, 1);
+        ASSERT_TRUE(_tc->isValid());
+        const YAML::Node& config = _tc->getConfig();
+        _spinner = std::make_unique<ros::AsyncSpinner>(config["Alica"]["ThreadPoolSize"].as<int>(4));
+        _tc->setCommunicator<alicaDummyProxy::AlicaDummyCommunication>();
+        _tc->setTimerFactory<alicaRosTimer::AlicaRosTimerFactory>();
+        _tc->setLogger<alicaRosLogger::AlicaRosLogger>(config["Local"]["ID"].as<int>());
+        _tc->setWorldModel<alicaTests::TestWorldModelNew>(_tc.get());
+        AlicaCreators creators{std::make_unique<alica::ConditionCreator>(), std::make_unique<alica::UtilityFunctionCreator>(),
+                std::make_unique<alica::ConstraintCreator>(), std::make_unique<alica::BehaviourCreator>(), std::make_unique<alica::PlanCreator>(),
+                std::make_unique<alica::TransitionConditionCreator>()};
+        _tc->init(std::move(creators));
+        _tc->startEngine();
+        _spinner->start();
+        STEP_UNTIL(_tc, _tc->getActivePlan("TestMasterPlan"));
+        ASSERT_TRUE(_tc->getActivePlan("TestMasterPlan")) << _tc->getLastFailure();
+    }
+
+    virtual void TearDown() override
+    {
+        _spinner->stop();
+        _tc->terminate();
+    }
+
+protected:
+    std::unique_ptr<TestContext> _tc;
+    std::unique_ptr<ros::AsyncSpinner> _spinner;
+};
+
+TEST_F(TestSuccessFixture, behSuccessTest)
+{
+    ASSERT_TRUE(_tc->setTransitionCond("TestMasterPlan", "ChooseTestState", "BehSuccessTestState")) << _tc->getLastFailure();
+    STEP_UNTIL(_tc, _tc->getActiveBehaviour("SuccessOnInitBeh"));
+    auto beh = _tc->getActiveBehaviour("SuccessOnInitBeh");
+    ASSERT_NE(beh, nullptr) << _tc->getLastFailure();
+    STEP_UNTIL(_tc, _tc->isSuccess(beh));
+    ASSERT_TRUE(_tc->isSuccess(beh));
+}
+
+TEST_F(TestSuccessFixture, planSuccessTest)
+{
+    ASSERT_TRUE(_tc->setTransitionCond("TestMasterPlan", "ChooseTestState", "PlanSuccessTestState")) << _tc->getLastFailure();
+    STEP_UNTIL(_tc, _tc->getActivePlan("SuccessOnInitPlan"));
+    auto plan = _tc->getActivePlan("SuccessOnInitPlan");
+    ASSERT_NE(plan, nullptr) << _tc->getLastFailure();
+    STEP_UNTIL(_tc, _tc->isSuccess(plan));
+    ASSERT_TRUE(_tc->isSuccess(plan));
```

Comment:
We should really introduce STEP_UNTIL_ASSERT_TRUE and STEP_UNTIL_ASSERT_EQ

Reply:
I think STEP_UNTIL_ASSERT_TRUE is sufficient since it can be used for all comparisons without having to add new STEP_UNTIL* macros for each type of comparison

Reply:
At least in gbc I find EQ is used more frequently that TRUE .  Everything else is not so common